### ISSUE TITLE
秒记

### ISSUE THREAD

**Issue by zhaipro** [zhaipro/ActionRecognition#1]
啊，该想办法洗牌啦，洗牌动作…

**Comment by zhaipro** [comment]
因为机器知道你鼓掌后就会用左手正面点击。

**Comment by zhaipro** [comment]
洗牌规则：对自然放下后的点进行切割，然后洗牌，\(^o^)/

**Comment by zhaipro** [comment]
简化学习难度

开始和开始_静止很容易识别，但是精确的区分还是有难度的。同样的还是有暂停和暂定_静止。
上翻_标准和上翻_自由有时候很难区分，因为有些人无论如何都很标准，而有些人…

不过，区分左右手似乎是一件小事儿。

我哪里搞错了吗？

**Comment by zhaipro** [comment]
动作与动作之间是关联的，机器看到你先是右手左滑，然后即使你放下手时有点像右手右滑，机器也能推倒出你应该在准备放下。

**Comment by zhaipro** [comment]
https://arxiv.org/abs/1803.01271

**Comment by zhaipro** [comment]
奥卡姆剃刀

**Comment by zhaipro** [comment]
数据集中有1.25%帧是缺少关键点的…
其原因是大幅度动作导致胳膊肘处境…
这有可能导致不佳的正则化。
尝试正则化尺寸，而不是x和y轴都正则化。
没有给我惊喜。

**Comment by zhaipro** [comment]
好想用到产品上体验一下。

**Comment by zhaipro** [comment]
今天奖赏一下自己，找到了曾经自己犯下的极低错误。

**Comment by zhaipro** [comment]
终于1.0版本诞生了。
开始，暂停，和左右滑动作识别效果很理想，可为什么上下翻动作识别效果就那么差呢？

**Comment by zhaipro** [comment]
一个动作与自然放下之间的情况我们可以将其定义为停顿。

**Comment by zhaipro** [comment]
只能通过实验来寻找问题了，我是干想不出来啦。

**Comment by zhaipro** [comment]
终于有想法了，重新定义权重。涨经验了。目前的最优策略是只给动作后期帧加权重，前期的完全扔掉。

因为准备上翻的动作本身就很像是左滑，所以我们还是不要给机器太大的压力了。

啊，对了，马上2.0版本就要诞生了

**Comment by zhaipro** [comment]
新的洗牌方案，交换左右胳膊，从而左滑改右滑。

**Comment by zhaipro** [comment]
这是一个动作识别，帧与帧之间应该有所关联。

**Comment by zhaipro** [comment]
关于动作识别我们很容易就能给机器打分，可是要如何定义成数学公式呢？

**Comment by zhaipro** [comment]
一个新的训练集将要到来，希望它更加的精准。初次接触，主观感受还是很不错的。
训练集来自于OpenPose。

**Comment by HeroBarry** [comment]
怎么玩？

**Comment by zhaipro** [comment]
@HeroBarry 还在开发阶段，准备用于体感游戏。

参考项目：https://github.com/philipperemy/keras-tcn

**Comment by HeroBarry** [comment]
嗯嗯，希望跟着学习一下，以后在智慧屏上做应用。

**Comment by zhaipro** [comment]
尝试把训练集中不用的动作统一定义成未知动作。

**Comment by zhaipro** [comment]
感觉我需要深入了解加权公式。

**Comment by zhaipro** [comment]
想办法让OpenPose支持多人游戏。

**Comment by zhaipro** [comment]
能不能直接训练模型来支持多人游玩呢？

**Comment by zhaipro** [comment]
Epoch 4000/4000
1/1 [=] - 0s 120ms/step - loss: 0.0071 - acc: 0.9821 - val_loss: 0.0097 - val_acc: 0.9726
